mllib/src/main/scala/org/apache/spark/ml/clustering/KMeans.scala

```diff
@@ -306,12 +307,19 @@ class KMeans @Since("1.5.0") (
 
   @Since("2.0.0")
   override def fit(dataset: Dataset[_]): KMeansModel = {
+    val handlePersistence = dataset.rdd.getStorageLevel == StorageLevel.NONE
+    fit(dataset, handlePersistence)
+  }
+  @Since("2.0.0")
```

Comment:
This still has several problems. This can't be since 2.0.0 and there is no space before this line.